lib/ansible/modules/web_infrastructure/ansible_tower/tower_workflow_template.py

```diff
@@ -0,0 +1,186 @@
+    state: absent
+'''
+
+
+RETURN = ''' # '''
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ansible_tower import (
+    TowerModule,
+    tower_auth_config,
+    tower_check_mode
+)
+
+try:
+    import tower_cli
+    import tower_cli.exceptions as exc
+    from tower_cli.conf import settings
+except ImportError:
+    pass
+
+
+def main():
+    argument_spec = dict(
+        name=dict(required=True),
+        description=dict(required=False),
+        extra_vars=dict(required=False),
+        organization=dict(required=False),
+        allow_simultaneous=dict(type='bool', required=False),
+        schema=dict(required=False),
+        survey=dict(required=False),
+        survey_enabled=dict(type='bool', required=False),
+        state=dict(choices=['present', 'absent'], default='present'),
+    )
+
+    module = TowerModule(
+        argument_spec=argument_spec,
+        supports_check_mode=False
+    )
+
+    name = module.params.get('name')
+    state = module.params.get('state')
+
+    json_output = {'workflow_template': name, 'state': state}
+
+    tower_auth = tower_auth_config(module)
+    with settings.runtime_values(**tower_auth):
+        tower_check_mode(module)
+        wfjt_res = tower_cli.get_resource('workflow')
+        try:
+            params = {}
+            params['name'] = name
+
+            if module.params.get('description'):
+                params['description'] = module.params.get('description')
+
+            if module.params.get('organization'):
+                organization_res = tower_cli.get_resource('organization')
+                try:
+                    organization = organization_res.get(
+                        name=module.params.get('organization'))
+                    params['organization'] = organization['id']
+                except exc.NotFound as excinfo:
+                    module.fail_json(
+                        msg='Failed to update organization source,'
+                        'organization not found: {0}'.format(excinfo),
+                        changed=False
+                    )
+
+            if module.params.get('survey'):
+                params['survey_spec'] = module.params.get('survey')
+
+            for key in ('allow_simultaneous', 'extra_vars', 'survey_enabled',
+                        'description'):
+                if module.params.get(key):
+                    params[key] = module.params.get(key)
+
+            if state == 'present':
+                params['create_on_missing'] = True
+                result = wfjt_res.modify(**params)
+                json_output['id'] = result['id']
+            elif state == 'absent':
+                params['fail_on_missing'] = False
+                result = wfjt_res.delete(**params)
+
+            if module.params.get('schema') and state == 'present':
+                wfjt_res.schema(
+                    result['id'],
+                    module.params.get('schema')
+                )
```

Comment:
Two things stroller me about this code: (1) merge it into the above if state == "present" block.  (2) you should have a check after the ansible module is created that errors if schema is set and state==absent

Reply:
1. I've relocated the schema association to the workflow template
2. I've added a condition for that